provisioning/baremetal_pod.go

```diff
@@ -278,34 +278,64 @@ func setIronicExternalIp(name string, config *metal3iov1alpha1.ProvisioningSpec)
 	}
 }
 
-func setIronicExternalUrl(client kubernetes.Interface, config *metal3iov1alpha1.ProvisioningSpec, namespace string) corev1.EnvVar {
-	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && config.VirtualMediaViaExternalNetwork {
-		ipv6PodIP, err := GetPodIP(client.CoreV1(), namespace, NetworkStackV6)
+func setIronicExternalUrl(info ProvisioningInfo) corev1.EnvVar {
+	// We need to set the external url when
+	//
+	//   1. IPv6 is enabled
+	//   2. VirtualMediaViaExternalNetwork is enabled
+	//
+	// When VirtualMediaViaExternalNetwork is enabled, ironic-proxy is also
+	// enabled.  This means that we can set the external url to be the IPv6
+	// version of ironic-proxy.
```

Comment:
This seems too restrictive on the face of it.

`VirtualMediaViaExternalNetwork` is for the specific case where there _is_ a provisioning network, but we _don't_ want to use it for virtualmedia deploys (i.e. provisioning network is only for PXE).
It seems like the proxy, and also this, should also be enabled in the case where there is _not_ a provisioning network. In that case there is only the external network and it goes without saying that all virtualmedia deployments go over that network.

BTW does the logic in BMO know to only set the alternative URL on the node for virtualmedia deployments? (I know it only does it when the BMC has an IPv6 address.) If not that would mess up PXE deployments when `VirtualMediaViaExternalNetwork` is enabled and the PXE network is IPv6.

Reply:
> BTW does the logic in BMO know to only set the alternative URL on the node for virtualmedia deployments? (I know it only does it when the BMC has an IPv6 address.) If not that would mess up PXE deployments when `VirtualMediaViaExternalNetwork` is enabled and the PXE network is IPv6.

It looks like Ironic does not use the `external_http_url` for PXE, so we should be ok.